docs/the-contribution-workflow.md

```diff
@@ -0,0 +1,245 @@
+
+Once it finishes downloading, change into the new directory, and congrats! You've got Exercism cloned down to your machine!
+
+
+3. Track the original Exercism repo
+
+As you might expect, the Exercism codebase changes constantly. It's important, therefore, to make sure you can easily stay up-to-date with new changes.
+
+To do that, you have to add the original Exercism repo as one of your remotes.
+
+To show what will happen, fell free to run `git remote -v` in your terminal.
+
+It should show something like this:
+
+```shell
+$ git remote -v
+origin	git@github.com:josh-works/exercism.io.git (fetch)
+origin	git@github.com:josh-works/exercism.io.git (push)
+```
+
+So, we want to add another "remote", pointing to https://github.com/exercism/exercism.io.git
+
+To do that, run:
+
+`git remote add upstream https://github.com/exercism/exercism.io.git`
+
+Now, when you run `git remote -v`, you should have _two_ remotes:
+
+```shell
+$ git remote -v
+
+origin	git@github.com:josh-works/exercism.io.git (fetch)
+origin	git@github.com:josh-works/exercism.io.git (push)
+upstream	https://github.com/exercism/exercism.io.git (fetch)
+upstream	https://github.com/exercism/exercism.io.git (push)
+```
+
+This is perfect. Now, go update your current repo with the 'upstream' repo, just run
+
+```shell
+git pull upstream master
+```
+
+And you'll update your master branch to the most current version from Exercism.
+
+
+# Working on the code
+
+OK, you've done quite a bit of work to get set up and ready to make your contributions.
+
+Next, you need to check out a new branch, and do the work.
+
+The name of the branch I'm working on right now, as I write this guide, is `contribution_for_beginners`.
+
+You might pick something similarly descriptive of the feature you're working on.
+
+So, with your checked out branch, do your work and make commits as usual. Add files, add descriptive commit messages, etc.
+
+
+
+# Submitting your contribution
+
+Once you've made your changes, you are ready to do one of two things:
+
+1. Submit them.
+2. Publish them for feedback
+
+The good news is the process for both of these steps is the same.
+
+You're going to make a "pull request", against the Exercism repo.
+
+[Here's a quick explanation/guide for pull requests](https://yangsu.github.io/pull-request-tutorial/), if you want some extra reading.
+
+## Making your pull request
+
+#### Add and commit your code
+
+This is the normal `git add <file>`, `git commit -m "your commit message"` bit.
+
+#### Fetch the current version of exercism, merge it into your branch
+
+Remember that Exercism is a huge codebase and changes regularly. So, you need to make sure that the code your working with is as up-to-date as possible.
+
+```shell
+# check out the master branch
+$ git checkout master
+
+# get changes to the exercism repo
+$ git fetch upstream master
+
+# merge your changes into your master branch
+$ git checkout your_working_branch
+$ git rebase -i your_working_branch master
```

Comment:
This is kind of technical, but merging is not rebasing. If you could describe all of this without saying "merge" that would make people who are good at Git twitch a lot less.